app/controllers/pro_controller.rb

```diff
@@ -6,18 +6,21 @@ def show
   def new
     @oauth_info = OauthParse.new(session[:omniauth_info])
     @services = Service.where(id: params[:service_id])
+    @radius = params[:radius]
+    session[:radius] = params[:radius]
```

Comment:
I'd try to send only one instance variable to the view for this form. Would be a good opportunity to use a presenter.